src/vm/runners/builtin_runner/mod.rs

```diff
@@ -303,17 +306,14 @@ impl BuiltinRunner {
             .into());
         }
 
-        // Since both offsets and this iterator are ordered, a simple pointer is
-        // enough to check if the values are present.
-        let mut offsets_iter = offsets.iter().copied().peekable();
         let mut missing_offsets = Vec::new();
         for i in 0..n as usize {
-            let offset = cells_per_instance as usize * i;
+            let offset_base = cells_per_instance as usize * i;
             for j in 0..n_input_cells as usize {
-                let offset = offset + j;
-                match offsets_iter.next_if_eq(&offset) {
-                    Some(_) => {}
-                    None => {
+                let offset = offset_base + j;
+                match offsets.contains(&offset) {
```

Comment:
This makes the complexity go from `O(n_input_cells)` to `O(n_input_cells*offsets.len()`. Could this be called often?

Reply:
I changed it back to `O(n*n_input_cells)`.